jbehave-core/src/test/java/org/jbehave/core/parsers/RegexStoryParserBehaviour.java

```diff
@@ -820,6 +819,30 @@ void shouldParseStoryWithScenarioKeywordInComments() {
                 "!-- 3 One more comment")));
     }
 
+    @Test
+    void shouldParseStoryWithNotFormattedComments() {
+        String wholeStory = "Scenario: with not formatted comments"
+                + NL + "!-- This is a comment"
```

Comment:
`"!-- This is a comment!--"` just in case :)

Reply:
there is no much sense for the current case, nevertheless I've added this test case as well